### ISSUE TITLE
Specify Language Version

### ISSUE THREAD

**Issue by moza88** [gitpod-io/workspace-images#821]
## Is your feature request related to a problem? Please describe

As we upgrade our code and experiment with different language versions l would like one Gitpod with one language version and another one with another language version. 

For instance, I can't use this tool called Kubebuilder with 1.18 go, I have to use 1.17 go. I have other projects that require 1.18, so switching back and forth locally between the versions can be cumbersome. 

## Describe the behavior you'd like

A way to specify in the gitpod.yml file which version of the language I would like to use, the configuration will override the default version. The default go version in gitpod is 1.18, I would mention in my gitpod.yml I want to use 1.17 and it would override it to make it 1.17. 

## Describe alternatives you've considered

The alternative is to use multiple IDEs on my computer to switch back and forth between language versions.

**Comment by axonasif** [comment]
Hey @moza88, we actually had multiple variants for `workspace-go` docker image which you could have used as `image: gitpod/workspace-go:1.17` on your `.gitpod.yml` to have your desired go version.

However, it was removed for not being used: https://github.com/gitpod-io/workspace-images/pull/761/files
@princerachit, maybe we can reconsider adding them?

But wait, you don't essentially need an `go-v1.17` image from us you, you can build one yourself! Which could make your life easier. Please see https://github.com/gitpod-io/website/blob/a0eb26f9e4ca2dcd5304d3b6db6c2f4e70aba379/gitpod/docs/languages/go.md#installing-custom-go-version-on-a-minimal-workspace

**Comment by tnir** [comment]
> Hey @moza88, we actually had multiple variants for workspace-go docker image which you could have used as image: gitpod/workspace-go:1.17 on your .gitpod.yml to have your desired go version.

@axonasif I do not think so. If you could have been able to do so, you can find that image at https://hub.docker.com/r/gitpod/workspace-go/tags?page=1&name=1.17 **unless intentionally delete it**.

#761 just deleted unused chunks...

**Comment by tnir** [comment]
> But wait, you don't essentially need an go-v1.17 image from us you, you can build one yourself in just a minute! Which could make your life easier. Please see https://github.com/gitpod-io/website/blob/a0eb26f9e4ca2dcd5304d3b6db6c2f4e70aba379/gitpod/docs/languages/go.md#installing-custom-go-version-on-a-minimal-workspace

@axonasif Installing go does not take a so longer time, but other languages do. Like Codespaces https://github.com/microsoft/vscode-dev-containers, why not adding more varieties in languages and middlewares?

Comment from community: https://discord.com/channels/816244985187008514/816246578594840586/983640504031932436

**Comment by axonasif** [comment]
@tnir yeah you're right, if it was, I would've told him to use that instead on his `.gitpod.yml` :)

**Comment by axonasif** [comment]
> @axonasif Installing go does not take a so longer time, but other languages do. Like Codespaces https://github.com/microsoft/vscode-dev-containers, why not adding more varieties in languages and middlewares?

That's something I'm not aware of, maybe @atduarte 🙏 ?

**Comment by atduarte** [comment]
@axonasif our goal is actually to allow users to select "layers" with additional languages/tools/software, but currently we don't have that capability nor the capacity to work on it.

Regarding this specific issue: we already have this for certain languages such as ruby, node and java [[example](https://github.com/gitpod-io/workspace-images/blob/main/dazzle.yaml#L135)]. We could create the same for golang specifically, but according to Google "It is intended that programs written to the Go 1 specification will continue to compile and run correctly, unchanged, over the lifetime of that specification.". Is this a rare corner case? Are there others having the same issue?
